appservice/src/createAppService/AppServicePlanCreateStep.ts

```diff
@@ -27,26 +28,47 @@ export class AppServicePlanCreateStep extends AzureWizardExecuteStep<IAppService
-
-        if (existingPlan) {
-            wizardContext.plan = existingPlan;
-            ext.outputChannel.appendLog(foundAppServicePlan);
-        } else {
-            ext.outputChannel.appendLog(creatingAppServicePlan);
-            progress.report({ message: creatingAppServicePlan });
-            const skuFamily = wizardContext.newPlanSku?.family ? wizardContext.newPlanSku?.family.toLowerCase() : '';
-            const isElasticPremiumOrWorkflowStandard: boolean = skuFamily === 'ep' || skuFamily === 'ws';
-            wizardContext.plan = await client.appServicePlans.createOrUpdate(rgName, newPlanName, {
-                kind: getAppServicePlanModelKind(wizardContext.newSiteKind, nonNullProp(wizardContext, 'newSiteOS')),
-                sku: nonNullProp(wizardContext, 'newPlanSku'),
-                location: nonNullValueAndProp(wizardContext.location, 'name'),
-                reserved: wizardContext.newSiteOS === WebsiteOS.linux,  // The secret property - must be set to true to make it a Linux plan. Confirmed by the team who owns this API.
-                maximumElasticWorkerCount: isElasticPremiumOrWorkflowStandard ? 20 : undefined
-            });
-            ext.outputChannel.appendLog(createdAppServicePlan);
+        try {
+            const client: WebSiteManagementClient = await createWebSiteClient(wizardContext);
+            const existingPlan: WebSiteManagementModels.AppServicePlan | undefined = await tryGetAppServicePlan(client, rgName, newPlanName);
+
+            if (existingPlan) {
+                wizardContext.plan = existingPlan;
+                ext.outputChannel.appendLog(foundAppServicePlan);
+            } else {
+                ext.outputChannel.appendLog(creatingAppServicePlan);
+                progress.report({ message: creatingAppServicePlan });
+                const skuFamily = wizardContext.newPlanSku?.family ? wizardContext.newPlanSku?.family.toLowerCase() : '';
+                const isElasticPremiumOrWorkflowStandard: boolean = skuFamily === 'ep' || skuFamily === 'ws';
+                wizardContext.plan = await client.appServicePlans.createOrUpdate(rgName, newPlanName, {
+                    kind: getAppServicePlanModelKind(wizardContext.newSiteKind, nonNullProp(wizardContext, 'newSiteOS')),
+                    sku: nonNullProp(wizardContext, 'newPlanSku'),
+                    location: nonNullValueAndProp(wizardContext.location, 'name'),
+                    reserved: wizardContext.newSiteOS === WebsiteOS.linux,  // The secret property - must be set to true to make it a Linux plan. Confirmed by the team who owns this API.
+                    maximumElasticWorkerCount: isElasticPremiumOrWorkflowStandard ? 20 : undefined
+                });
+                ext.outputChannel.appendLog(createdAppServicePlan);
+            }
+        } catch (e) {
+            if (wizardContext.suppress403Handling || parseError(e).errorType !== '403') {
```

Comment:
I never meant `suppress403Handling` to apply this widely - it was supposed to be unique to resource groups. The only reason I needed it was because in the rg extension [here](https://github.com/microsoft/vscode-azureresourcegroups/blob/b70b91ff4765f9ee861a4325b363b2333334296f/src/tree/SubscriptionTreeItem.ts#L39), the user is explicitly trying to create a resource group, so it would make no sense to "select an existing".

The correct fix is probably for me to move `suppress403Handling` from the `wizardContext` to a parameter in the constructor of `ResourceGroupCreateStep`. Not sure when I'll get to that, but for now I suggest you just ignore `suppress403Handling`